cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -280,6 +280,70 @@ void generate_depth_remappings(std::map<int, std::pair<std::vector<int>, std::ve
   return total_pages;
 }
 
+/**
+ * @brief Returns a string representation of known encodings
+ *
+ * @param encoding Given encoding
+ * @return STring representation of encoding
+ */
+std::string encoding_to_string(Encoding encoding)
+{
+  switch (encoding) {
+    case Encoding::PLAIN: return "PLAIN";
+    case Encoding::GROUP_VAR_INT: return "GROUP_VAR_INT";
+    case Encoding::PLAIN_DICTIONARY: return "PLAIN_DICTIONARY";
+    case Encoding::RLE: return "RLE";
+    case Encoding::BIT_PACKED: return "BIT_PACKED";
+    case Encoding::DELTA_BINARY_PACKED: return "DELTA_BINARY_PACKED";
+    case Encoding::DELTA_LENGTH_BYTE_ARRAY: return "DELTA_LENGTH_BYTE_ARRAY";
+    case Encoding::DELTA_BYTE_ARRAY: return "DELTA_BYTE_ARRAY";
+    case Encoding::RLE_DICTIONARY: return "RLE_DICTIONARY";
+    case Encoding::BYTE_STREAM_SPLIT: return "BYTE_STREAM_SPLIT";
+    case Encoding::NUM_ENCODINGS:
+    default: return "UNKNOWN";
+  }
+}
+
+/**
+ * @brief Helper function to convert an encoding bitmask to a readable string
+ *
+ * @param bitmask Bitmask of found unsupported encodings
+ * @returns Human readable string with unsupported encodings
+ */
+std::string encoding_bitmask_to_str(int32_t encoding_bitmask)
+{
+  std::string result;
+  for (int32_t i = 0; i < 32; ++i) {
+    if (encoding_bitmask & (1 << i)) {
+      Encoding current = static_cast<Encoding>(i);
+      if (!is_supported_encoding(current)) { result.append(encoding_to_string(current) + " "); }
+    }
+  }
+  return result;
+}
+
+/**
+ * @brief Create a readable string for the user that will list out all unsupported encodings found.
+ *
+ * @param pages List of page information
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @returns Human readable string with unsupported encodings
+ */
+std::string list_unsupported_encodings(const cudf::detail::hostdevice_vector<PageInfo>& pages,
+                                       rmm::cuda_stream_view stream)
+{
+  auto to_mask = [] __device__(const PageInfo& page) -> int8_t {
+    return is_supported_encoding(page.encoding) ? 0U : encoding_to_mask(page.encoding);
+  };
+  int32_t unsupported = thrust::transform_reduce(rmm::exec_policy(stream),
```

Comment:
```suggestion
  uint32_t const unsupported = thrust::transform_reduce(rmm::exec_policy(stream),
```
I too wish C++ had Scala's `val` vs `var` :sweat_smile:

Reply:
Ahhh my bad. I context switch too much so the habits from both languages don't translate well😂. All good. I appreciate the review! :)